common/src/TaskCompletionSource.cs

```diff
@@ -5,30 +5,22 @@
 
 namespace Microsoft.Azure.Devices.Shared
 {
-    internal sealed class TaskCompletionSource : TaskCompletionSource<int>
+    /// <summary>
+    /// A <see cref="TaskCompletionSource{boolean}"/> implementation that returns a <see cref="Task"/> when completed.
+    /// </summary>
+    /// <remarks>
+    /// This is used for .NET implementations less than .NET 5.0, which lack a native implementation of the non-generic TaskCompletionSource.
+    /// </remarks>
+    internal sealed class TaskCompletionSource : TaskCompletionSource<bool>
     {
-        public static readonly TaskCompletionSource Void = CreateVoidTcs();
-
-        public TaskCompletionSource(object state)
-            : base(state)
-        {
-        }
-
         public TaskCompletionSource()
         {
         }
 
-        public bool TrySetResult() => TrySetResult(0);
+        public bool TrySetResult() => TrySetResult(true);
```

Comment:
Should we set this to true or false? Looks like it was 0 (false) previeously.

Reply:
It doesn't really matter, all it does it mark the task as completed. We could have also defined it as an object and set it to null.